framework/src/meshgenerators/FancyExtruderGenerator.C

```diff
@@ -0,0 +1,547 @@
+            new_elem->set_node(25) =
+                mesh->node_ptr(elem->node_ptr(8)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(26) =
+                mesh->node_ptr(elem->node_ptr(8)->id() + ((2 * current_layer + 1) * orig_nodes));
+
+            if (elem->neighbor_ptr(0) == remote_elem)
+              new_elem->set_neighbor(1, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(1) == remote_elem)
+              new_elem->set_neighbor(2, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(2) == remote_elem)
+              new_elem->set_neighbor(3, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(3) == remote_elem)
+              new_elem->set_neighbor(4, const_cast<RemoteElem *>(remote_elem));
+
+            break;
+          }
+          default:
+          {
+            libmesh_not_implemented();
+            break;
+          }
+        }
+
+        new_elem->set_id(elem->id() + (current_layer * orig_elem));
+        new_elem->processor_id() = elem->processor_id();
+
+#ifdef LIBMESH_ENABLE_UNIQUE_ID
+        // Let's give the base of the extruded mesh the same
+        // unique_ids as the source mesh, in case anyone finds that
+        // a useful map to preserve.
+        const unique_id_type uid = (current_layer == 0)
+                                       ? elem->unique_id()
+                                       : orig_unique_ids +
+                                             (current_layer - 1) * (orig_nodes + orig_elem) +
+                                             orig_nodes + elem->id();
+
+        new_elem->set_unique_id() = uid;
+#endif
+
+        // maintain the subdomain_id
+        new_elem->subdomain_id() = elem->subdomain_id();
+
+        if (_subdomain_swap_pairs.size())
+        {
+          auto & elevation_swap_pairs = _subdomain_swap_pairs[e];
+
+          auto new_id_it = elevation_swap_pairs.find(elem->subdomain_id());
+
+          if (new_id_it != elevation_swap_pairs.end())
+            new_elem->subdomain_id() = new_id_it->second;
+        }
+
+        new_elem = mesh->add_elem(new_elem);
+
+        // Copy any old boundary ids on all sides
+        for (auto s : elem->side_index_range())
+        {
+          input_boundary_info.boundary_ids(elem, s, ids_to_copy);
+
+          if (new_elem->dim() == 3)
+          {
+            // For 2D->3D extrusion, we give the boundary IDs
+            // for side s on the old element to side s+1 on the
+            // new element.  This is just a happy coincidence as
+            // far as I can tell...
+            boundary_info.add_side(new_elem, cast_int<unsigned short>(s + 1), ids_to_copy);
+          }
+          else
+          {
+            // For 1D->2D extrusion, the boundary IDs map as:
+            // Old elem -> New elem
+            // 0        -> 3
+            // 1        -> 1
+            libmesh_assert_less(s, 2);
```

Comment:
I know you copied this code. Do we want to keep the libmesh asserts? I guess they are fine.